### PR TITLE
Fix #1421

### LINKED ISSUE #1421: method for relocating entire library
I was wondering if there's a way for the beets db to use relative paths?

I find myself moving to a new machine with a new username, and thus a new absolute path for my music library..
If there is no way to use relative paths in the db, how would you suggest updating the db?

(BTW) I contain ALL my beets-managed music in one directory per beets db, so this should be straightforward.  Perhaps, this should be in the docs?

### PR DESCRIPTION
**Pull request by shanemikel**
Just a convenience.

Checks the db for paths that can't be found in the fs, and fixes them to match Item.destination() if a file exists at that path.

### CODE REVIEW

**Comment by sampsyo** [comment]
Interesting idea! This could indeed solve the case where people move files before informing beets. A few to-do items:

* I think this should live in a plugin—not for a technical reason, but for documentation purposes, this will make it clearer that this is not functionality meant to be used on a day-to-day basis.
* Travis has reported a few style issues.
* We'll need to add documentation.

Thanks for the contribution!

File: beets/ui/commands.py [hunk 1]
```diff
@@ -1395,6 +1395,55 @@ def move_func(lib, opts, args):
...
+        if os.path.exists(item.destination()):
+            found.append(item)
+
+    plural = not_found != 1
+    log.info(u'{0} item' + (u's' if plural else u'') + u' in the database have'
+            u' bad paths, and {1} will be fixed.', len(not_found), len(found))
+    
+    unchanged = len(not_found) - len(found)
+    if unchanged:
+        log.info(u'{0} must be migrated manually.', unchanged)
```
Comment (sampsyo) [inline beets/ui/commands.py]: For both this log message and the one above, a terser style would fit better with the beets style: something like:

    found 9 bad paths, can migrate 7
    not migrated: /some/path.mp3
    not migrated: /another/path.mp3

Or even:

    found 9 bad paths
    2 not migrated:
    /some/path.mp3
    /another/path.mp3

**Comment by shanemikel** [comment]
Do you see a place for options like --old-path-prefix/--new-path-prefix?

**Comment by sampsyo** [comment]
Hmm, not sure—seems fine without explicit paths at the moment.

**Comment by shanemikel** [comment]
I'll work it into a plugin, but it'll be a few days until I get around to learning the plugin architecture.

**Comment by jtpavlock** [comment]
closing as abandoned

**Pull request closed without merging**